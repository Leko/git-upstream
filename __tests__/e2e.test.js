const { spawnSync } = require('child_process')

describe('E2E', () => {
    it('--help', () => {
        const { status, output: [, stdout, stderr] } = spawnSync('node', ['../index.js', '--help'])
        expect(status).toBe(0)
        expect({ stdout, stderr }).toMatchSnapshot()
    })
    it('--set', () => {
        const { status, output: [, stdout, stderr] } = spawnSync('node', ['../index.js', '--set'])
        expect(status).toBe(0)
        expect({ stdout, stderr }).toMatchSnapshot()
    })
    it('--set -f', () => {
        const { status, output: [, stdout, stderr] } = spawnSync('node', ['../index.js', '--set', '-f'])
        expect(status).toBe(0)
        expect({ stdout, stderr }).toMatchSnapshot()
    })
    it('--set -f --https', () => {
        const { status, output: [, stdout, stderr] } = spawnSync('node', ['../index.js', '--set', '-f', '--https'])
        expect(status).toBe(0)
        expect({ stdout, stderr }).toMatchSnapshot()
    })
    it('--set -f --no-shallow', () => {
        const { status, output: [, stdout, stderr] } = spawnSync('node', ['../index.js', '--set', '-f', '--no-shallow'])
        expect(status).toBe(0)
        expect({ stdout, stderr }).toMatchSnapshot()
    })
})
