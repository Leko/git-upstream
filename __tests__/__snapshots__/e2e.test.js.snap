// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E --help 1`] = `
Object {
  "stderr": Object {
    "data": Array [],
    "type": "Buffer",
  },
  "stdout": Object {
    "data": Array [
      79,
      112,
      116,
      105,
      111,
      110,
      115,
      58,
      10,
      32,
      32,
      45,
      45,
      104,
      101,
      108,
      112,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      83,
      104,
      111,
      119,
      32,
      104,
      101,
      108,
      112,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      91,
      98,
      111,
      111,
      108,
      101,
      97,
      110,
      93,
      10,
      32,
      32,
      45,
      45,
      118,
      101,
      114,
      115,
      105,
      111,
      110,
      32,
      32,
      32,
      32,
      83,
      104,
      111,
      119,
      32,
      118,
      101,
      114,
      115,
      105,
      111,
      110,
      32,
      110,
      117,
      109,
      98,
      101,
      114,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      91,
      98,
      111,
      111,
      108,
      101,
      97,
      110,
      93,
      10,
      32,
      32,
      45,
      45,
      115,
      101,
      116,
      44,
      32,
      45,
      105,
      32,
      32,
      32,
      32,
      83,
      101,
      116,
      32,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      116,
      111,
      32,
      97,
      32,
      114,
      101,
      109,
      111,
      116,
      101,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      91,
      98,
      111,
      111,
      108,
      101,
      97,
      110,
      93,
      32,
      91,
      100,
      101,
      102,
      97,
      117,
      108,
      116,
      58,
      32,
      102,
      97,
      108,
      115,
      101,
      93,
      10,
      32,
      32,
      45,
      45,
      102,
      111,
      114,
      99,
      101,
      44,
      32,
      45,
      102,
      32,
      32,
      70,
      111,
      114,
      99,
      101,
      32,
      111,
      118,
      101,
      114,
      119,
      114,
      105,
      116,
      101,
      32,
      101,
      118,
      101,
      110,
      32,
      105,
      102,
      32,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      97,
      108,
      114,
      101,
      97,
      100,
      121,
      32,
      101,
      120,
      105,
      115,
      116,
      115,
      10,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      91,
      98,
      111,
      111,
      108,
      101,
      97,
      110,
      93,
      32,
      91,
      100,
      101,
      102,
      97,
      117,
      108,
      116,
      58,
      32,
      102,
      97,
      108,
      115,
      101,
      93,
      10,
      32,
      32,
      45,
      45,
      116,
      111,
      107,
      101,
      110,
      44,
      32,
      45,
      116,
      32,
      32,
      71,
      105,
      116,
      72,
      117,
      98,
      32,
      97,
      99,
      99,
      101,
      115,
      115,
      32,
      116,
      111,
      107,
      101,
      110,
      32,
      102,
      111,
      114,
      32,
      112,
      114,
      105,
      118,
      97,
      116,
      101,
      32,
      114,
      101,
      112,
      111,
      115,
      105,
      116,
      111,
      114,
      121,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      91,
      115,
      116,
      114,
      105,
      110,
      103,
      93,
      10,
      32,
      32,
      45,
      45,
      115,
      104,
      97,
      108,
      108,
      111,
      119,
      32,
      32,
      32,
      32,
      80,
      114,
      101,
      102,
      101,
      114,
      32,
      112,
      97,
      114,
      101,
      110,
      116,
      32,
      102,
      111,
      114,
      107,
      40,
      100,
      101,
      102,
      97,
      117,
      108,
      116,
      41,
      32,
      111,
      114,
      32,
      97,
      110,
      99,
      101,
      115,
      116,
      111,
      114,
      32,
      102,
      111,
      114,
      107,
      10,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      91,
      98,
      111,
      111,
      108,
      101,
      97,
      110,
      93,
      32,
      91,
      100,
      101,
      102,
      97,
      117,
      108,
      116,
      58,
      32,
      116,
      114,
      117,
      101,
      93,
      10,
      32,
      32,
      45,
      45,
      104,
      116,
      116,
      112,
      115,
      32,
      32,
      32,
      32,
      32,
      32,
      83,
      101,
      116,
      32,
      104,
      116,
      116,
      112,
      115,
      32,
      117,
      114,
      108,
      32,
      116,
      111,
      32,
      114,
      101,
      109,
      111,
      116,
      101,
      44,
      32,
      111,
      116,
      104,
      101,
      114,
      119,
      105,
      115,
      101,
      32,
      115,
      101,
      116,
      32,
      115,
      115,
      104,
      32,
      117,
      114,
      108,
      10,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      91,
      98,
      111,
      111,
      108,
      101,
      97,
      110,
      93,
      32,
      91,
      100,
      101,
      102,
      97,
      117,
      108,
      116,
      58,
      32,
      102,
      97,
      108,
      115,
      101,
      93,
      10,
      10,
      69,
      120,
      97,
      109,
      112,
      108,
      101,
      115,
      58,
      10,
      32,
      32,
      110,
      112,
      120,
      32,
      103,
      105,
      116,
      45,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      83,
      104,
      111,
      119,
      32,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      117,
      114,
      108,
      10,
      32,
      32,
      110,
      112,
      120,
      32,
      103,
      105,
      116,
      45,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      45,
      45,
      115,
      101,
      116,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      83,
      101,
      116,
      32,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      119,
      105,
      116,
      104,
      32,
      99,
      111,
      110,
      102,
      105,
      114,
      109,
      97,
      116,
      105,
      111,
      110,
      10,
      32,
      32,
      110,
      112,
      120,
      32,
      103,
      105,
      116,
      45,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      45,
      45,
      115,
      101,
      116,
      32,
      45,
      102,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      83,
      101,
      116,
      32,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      101,
      118,
      101,
      110,
      32,
      105,
      102,
      32,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      97,
      108,
      114,
      101,
      97,
      100,
      121,
      10,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      101,
      120,
      105,
      115,
      116,
      115,
      10,
      32,
      32,
      110,
      112,
      120,
      32,
      103,
      105,
      116,
      45,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      45,
      45,
      115,
      101,
      116,
      32,
      45,
      45,
      104,
      116,
      116,
      112,
      115,
      32,
      32,
      32,
      32,
      32,
      32,
      32,
      83,
      101,
      116,
      32,
      104,
      116,
      116,
      112,
      115,
      58,
      47,
      47,
      46,
      46,
      46,
      10,
      32,
      32,
      110,
      112,
      120,
      32,
      103,
      105,
      116,
      45,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      45,
      45,
      115,
      101,
      116,
      32,
      45,
      45,
      110,
      111,
      45,
      115,
      104,
      97,
      108,
      108,
      111,
      119,
      32,
      32,
      70,
      111,
      114,
      32,
      100,
      101,
      101,
      112,
      32,
      110,
      101,
      115,
      116,
      101,
      100,
      32,
      102,
      111,
      114,
      107,
      10,
    ],
    "type": "Buffer",
  },
}
`;

exports[`E2E --set -f --https 1`] = `
Object {
  "stderr": Object {
    "data": Array [],
    "type": "Buffer",
  },
  "stdout": Object {
    "data": Array [
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      105,
      115,
      32,
      104,
      116,
      116,
      112,
      115,
      58,
      47,
      47,
      103,
      105,
      116,
      104,
      117,
      98,
      46,
      99,
      111,
      109,
      47,
      110,
      111,
      100,
      101,
      106,
      115,
      47,
      110,
      111,
      100,
      101,
      46,
      103,
      105,
      116,
      10,
      65,
      100,
      100,
      32,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      116,
      111,
      32,
      97,
      32,
      114,
      101,
      109,
      111,
      116,
      101,
      10,
    ],
    "type": "Buffer",
  },
}
`;

exports[`E2E --set -f --no-shallow 1`] = `
Object {
  "stderr": Object {
    "data": Array [],
    "type": "Buffer",
  },
  "stdout": Object {
    "data": Array [
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      105,
      115,
      32,
      103,
      105,
      116,
      64,
      103,
      105,
      116,
      104,
      117,
      98,
      46,
      99,
      111,
      109,
      58,
      110,
      111,
      100,
      101,
      106,
      115,
      47,
      110,
      111,
      100,
      101,
      46,
      103,
      105,
      116,
      10,
      65,
      100,
      100,
      32,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      116,
      111,
      32,
      97,
      32,
      114,
      101,
      109,
      111,
      116,
      101,
      10,
    ],
    "type": "Buffer",
  },
}
`;

exports[`E2E --set -f 1`] = `
Object {
  "stderr": Object {
    "data": Array [],
    "type": "Buffer",
  },
  "stdout": Object {
    "data": Array [
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      105,
      115,
      32,
      103,
      105,
      116,
      64,
      103,
      105,
      116,
      104,
      117,
      98,
      46,
      99,
      111,
      109,
      58,
      110,
      111,
      100,
      101,
      106,
      115,
      47,
      110,
      111,
      100,
      101,
      46,
      103,
      105,
      116,
      10,
      65,
      100,
      100,
      32,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      116,
      111,
      32,
      97,
      32,
      114,
      101,
      109,
      111,
      116,
      101,
      10,
    ],
    "type": "Buffer",
  },
}
`;

exports[`E2E --set 1`] = `
Object {
  "stderr": Object {
    "data": Array [],
    "type": "Buffer",
  },
  "stdout": Object {
    "data": Array [
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      105,
      115,
      32,
      103,
      105,
      116,
      64,
      103,
      105,
      116,
      104,
      117,
      98,
      46,
      99,
      111,
      109,
      58,
      110,
      111,
      100,
      101,
      106,
      115,
      47,
      110,
      111,
      100,
      101,
      46,
      103,
      105,
      116,
      10,
      65,
      100,
      100,
      32,
      117,
      112,
      115,
      116,
      114,
      101,
      97,
      109,
      32,
      116,
      111,
      32,
      97,
      32,
      114,
      101,
      109,
      111,
      116,
      101,
      10,
    ],
    "type": "Buffer",
  },
}
`;
